### PR TITLE
MipsAsmParser/O32: Don't add redundant $ to $-prefixed symbol in the la macro

### PR DESCRIPTION
**Pull request by wzssyqa**
When parsing the `la` macro, we add a duplicate `$` prefix in `getOrCreateSymbol`,
leading to `error: Undefined temporary symbol $$yy` for code like:

```
xx:
	la	$2,$yy
$yy:
	nop
```

Remove the duplicate prefix.

In addition, recognize `.L`-prefixed symbols as local for O32.

See: #65020.

### CODE REVIEW

**Comment by wzssyqa** [comment]
@MaskRay @brad0

File: llvm/test/MC/Mips/macro-la-local.s [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# RUN: llvm-mc %s -triple=mips-unknown-linux -show-encoding -mcpu=mips32r2 | \
+# RUN:   FileCheck %s --check-prefixes=CHECK
+# RUN: llvm-mc %s -triple=mips-unknown-linux -show-encoding -mcpu=mips32r6 | \
+# RUN:   FileCheck %s --check-prefixes=CHECK
+
+	.text
+	.abicalls
+	.option	pic2
+xx:
+	la	$2,.Lhello #CHECK: lw      $2, %got(.Lhello)($gp)          # encoding: [0x8f,0x82,A,A]
```
Comment (MaskRay) [inline llvm/test/MC/Mips/macro-la-local.s:10]:
```
# CHECK:      lw ...
# CHECK-NEXT:   fixup A ...
# CHECK-NEXT:   ...
la $2, $hello2
```

File: llvm/lib/Target/Mips/AsmParser/MipsAsmParser.cpp [hunk 2]
```diff
@@ -2920,6 +2920,11 @@ bool MipsAsmParser::loadAndAddSymbolAddress(const MCExpr *SymExpr,
         (Res.getSymA()->getSymbol().isELF() &&
          cast<MCSymbolELF>(Res.getSymA()->getSymbol()).getBinding() ==
              ELF::STB_LOCAL);
+    if (!IsLocalSym && ABI.IsO32()) {
+      // PrivateGlobalPrefix for O32 is '$', while we support '.L' anyway.
```
Comment (MaskRay) [inline llvm/lib/Target/Mips/AsmParser/MipsAsmParser.cpp:2924]: This comment is confusing.

// For O32, "$"-prefixed symbols are recognized as temporary while .L-prefixed symbols are not (PrivateGlobalPrefix is "$"). Recognize ".L" manually.

File: llvm/test/MC/Mips/macro-la-local.s [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# RUN: llvm-mc %s -triple=mips-unknown-linux -show-encoding -mcpu=mips32r2 | \
```
Comment (MaskRay) [inline llvm/test/MC/Mips/macro-la-local.s:1]: remove `-unknown-linux` whenever applicable, to emphasize that this is generic ELF behavior.

**Review by MaskRay — APPROVED** [review]

**Comment by MaskRay** [comment]
Suggested description:

When parsing the `la` macro, we add a duplicate `$` prefix in `getOrCreateSymbol`,
leading to `error: Undefined temporary symbol $$yy` for code like:

```
xx:
	la	$2,$yy
$yy:
	nop
```

Remove the duplicate prefix.

In addition, recognize `.L`-prefixed symbols as local for O32.

**Pull request merged**